### PR TITLE
[API] Product Options

### PR DESCRIPTION
**Pull request by Rafikooo**
| Q               | A
|-----------------|-----
| Branch?         | api-platform-3 
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no 
| License         | MIT

Before:
image
After:
image

### CODE REVIEW

File: tests/Api/Admin/ProductOptionsTest.php [hunk 1]
```diff
@@ -244,14 +244,16 @@ public function it_does_not_update_a_product_option_value_with_duplicate_locale_
         /** @var ProductOptionValueInterface $productOptionValue */
         $productOptionValue = $fixtures['product_option_value_color_blue'];
 
+        $productOptionCode = $productOption->getCode();
+
         $this->client->request(
             method: 'PUT',
-            uri: sprintf('/api/v2/admin/product-options/%s', $productOption->getCode()),
+            uri: '/api/v2/admin/product-options/' . $productOptionCode,
```
Comment (mpysiak) [inline tests/Api/Admin/ProductOptionsTest.php:251]:
```suggestion
            uri: sprintf('/api/v2/admin/product-options/%s', $productOptionCode),
```

**Review by mpysiak — APPROVED** [review]

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/ProductOptionValue.xml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+-->
+
+<resources
+    xmlns="https://api-platform.com/schema/metadata/resources-3.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="https://api-platform.com/schema/metadata/resources-3.0 https://api-platform.com/schema/metadata/resources-3.0.xsd"
+>
+    <resource class="Sylius\Component\Product\Model\ProductOptionValue">
+        <operations>
+            <operation class="ApiPlatform\Metadata\NotExposed" uriTemplate="/admin/product-options/{product_option_code}/values/{code}">
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/ProductOptionValue.xml:21]:
```suggestion
            <operation class="ApiPlatform\Metadata\NotExposed" uriTemplate="/admin/product-options/{productOptionCode}/values/{code}">
```
or even better
```suggestion
            <operation class="ApiPlatform\Metadata\NotExposed" uriTemplate="/admin/product-options/{optionCode}/values/{code}">
```

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/ProductOptionValue.xml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+                <uriVariables>
+                    <uriVariable parameterName="product_option_code" fromClass="Sylius\Component\Product\Model\ProductOption" fromProperty="values"/>
+                    <uriVariable parameterName="code" fromClass="Sylius\Component\Product\Model\ProductOptionValue" />
+                </uriVariables>
+            </operation>
+
+            <operation class="ApiPlatform\Metadata\NotExposed" uriTemplate="/shop/product-options/{product_option_code}/values/{code}">
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/ProductOptionValue.xml:28]: As above

**Review by GSadee — APPROVED** [review]

File: tests/Api/Admin/ProductOptionsTest.php (hunk 1 above)
Comment (Rafikooo) [reply tests/Api/Admin/ProductOptionsTest.php:251]: https://github.com/Sylius/Sylius/pull/16707

**Pull request merged**